### PR TITLE
Add retire_old_submissions command

### PR DESCRIPTION
**Pull request by jibsheet**
The old command was "retire failed submissions" so I renamed it

New command requires a queue and allows you to also only retire old submissions.

OPS-3057

### CODE REVIEW

File: queue/management/commands/retire_failed_submissions.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+class Command(BaseCommand):
+    help = """
+           Retire submissions that have more than settings.MAX_NUMBER_OF_FAILURES failures.\
+           Notify the LMS/student that the queue will no longer attempt to process the submission.\
+           Optional <queue_name>
+           """
+
+    def add_arguments(self, parser):
+        parser.add_argument('queue_name', nargs='*')
```
Comment (feanil) [inline queue/management/commands/retire_failed_submissions.py]: Add a help message saying that if no 'queue_name's are provided, then we will retire submissions from all queues.

File: queue/management/commands/retire_submissions.py [hunk 2]
```diff
@@ -1,59 +1,47 @@
...
-            default=False,
-            help='Force retire submissions',
-        )
+            'queue_name',
+            help="Which queue to retire submissions for")
+
+        parser.add_argument(
+            '--retire-before',
+            dest='retire_before',
+            help='Date + time to limit submissions up for retirement (arrival_time < YYYY-MM-DD HH:MM::SS)'
```
Comment (feanil) [inline queue/management/commands/retire_submissions.py]: Also mention that this time should be in UTC.

File: queue/management/commands/retire_submissions.py [hunk 3]
```diff
@@ -1,59 +1,47 @@
 import logging
```
Comment (feanil) [inline queue/management/commands/retire_submissions.py]: Maybe rename this command to `retire_old_submissions.py` for clarity?

**Review by feanil — APPROVED** [review]
Couple of nits, but generally looks good.

File: queue/management/commands/retire_failed_submissions.py (hunk 1 above)
Comment (jibsheet) [reply queue/management/commands/retire_failed_submissions.py]: This is one of "positional - need a queue" args handled by management commands, seemed honestly more likely to me that you'd be retiring a single queue and I didn't want to accidentally do them all.
It'll error if you don't pass in a queue_name

File: queue/management/commands/retire_submissions.py (hunk 2 above)
Comment (jibsheet) [reply queue/management/commands/retire_submissions.py]: Good point - at one point that was here, but was lost when I cleaned up what style of date you needed

File: queue/management/commands/retire_submissions.py (hunk 3 above)
Comment (jibsheet) [reply queue/management/commands/retire_submissions.py]: Sure

File: queue/management/commands/retire_failed_submissions.py (hunk 1 above)
Comment (feanil) [reply queue/management/commands/retire_failed_submissions.py]: If that's the case why do you have the if statement below that will try to retire all submissions if no queue_name is passed?  Should that get cleaned up?

Comment (jibsheet) [reply queue/management/commands/retire_failed_submissions.py]: Oh, sorry, I misread the name of the file and the diff was too tight to make it obvious.
This is the old command.  It does treat queue_name as optional.  My newer one treats it as "please pass one or Django /argparse will throw an error".
I'll add something to the help statement above.

**Pull request merged**